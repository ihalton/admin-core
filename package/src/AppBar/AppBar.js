import React, { Children } from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import {
  AppBar as MuiAppBar,
  Box,
  Hidden,
  IconButton,
  Toolbar,
  Typography,
  makeStyles
} from "@material-ui/core";
import MenuIcon from "mdi-material-ui/Menu";
import ArrowLeftIcon from "mdi-material-ui/ArrowLeft";
import { UIContext } from "../context/UIContext";

const useStyles = makeStyles((theme) => ({
  action: {
    marginLeft: theme.spacing()
  },
  primarySidebarOpen: {
    ...theme.mixins.leadingPaddingWhenPrimaryDrawerIsOpen
  },
  detailDrawerOpen: {
    ...theme.mixins.trailingPaddingWhenDetailDrawerIsOpen
  },
  title: {
    flex: 1
  }
}));

/**
 * An AppBar for the main content area that provides a place for a title,
 * actions to the right, and a menu button for opening and closing the sidebar drawer menu
 * @param {Object} props Component props
 * @returns {React.Component} A react component
 */
function AppBar({ children, title, onBackButtonClick, shouldShowBackButton = true }) {
  const classes = useStyles();

  return (
    <UIContext.Consumer>
      {({ isMobile, isDetailDrawerOpen, isPrimarySidebarOpen, onTogglePrimarySidebar }) => {
        const toolbarClassName = clsx({
          // Add padding to the left when the primary sidebar is open,
          // only if we're on desktop. On mobile the sidebar floats over
          // the content like a modal that's docked to either the left
          // or right side of the screen.
          [classes.primarySidebarOpen]: isPrimarySidebarOpen && !isMobile,

          // Add padding to the right when the detail sidebar is open.
          // Omit on mobile as the sidebar will float over content.
          [classes.detailDrawerOpen]: isDetailDrawerOpen && !isMobile
        });

        return (
          <MuiAppBar>
            <Toolbar className={toolbarClassName}>
              <Hidden mdUp>
                <IconButton onClick={onTogglePrimarySidebar}>
                  <MenuIcon />
                </IconButton>
              </Hidden>
              {(shouldShowBackButton && onBackButtonClick) && (
                <Box paddingRight={1}>
                  <IconButton onClick={onBackButtonClick}>
                    <ArrowLeftIcon />
                  </IconButton>
                </Box>
              )}
              <Typography
                className={classes.title}
                component="h1"
                variant="h3"
              >
                {title}
              </Typography>
              {Children.map(children, (child) => (
                <div className={classes.action}>
                  {child}
                </div>
              ))}
            </Toolbar>
          </MuiAppBar>
        );
      }}
    </UIContext.Consumer>
  );
}

AppBar.propTypes = {
  children: PropTypes.node,
  onBackButtonClick: PropTypes.func,
  onToggleDrawerOpen: PropTypes.func,
  shouldShowBackButton: PropTypes.bool,
  title: PropTypes.node
};

export default AppBar;
