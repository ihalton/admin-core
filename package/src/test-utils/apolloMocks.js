import getViewer from "../graphql/queries/viewer.graphql";

export default [
  {
    request: {
      query: getViewer
    },
    result: {
      data: {
        viewer: {
          _id: "1234"
        }
      }
    }
  }
];
