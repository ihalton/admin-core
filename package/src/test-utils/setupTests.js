// // Add async/await features
// import "regenerator-runtime/runtime";

// // add some helpful assertions
import "@testing-library/jest-dom/extend-expect";

// // Polyfill fetch
// import "unfetch/polyfill";
