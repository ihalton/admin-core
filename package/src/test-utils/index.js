
import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/react-testing";
import { defaultTheme } from "@reactioncommerce/catalyst";
import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import apolloMocks from "./apolloMocks";

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: {}
      }
    },
    lng: "en",
    fallbackLng: "en",

    interpolation: {
      escapeValue: false
    }
  });

/**
 * Component that wraps components with mock providers during testing.
 * @return {Component} - Component wrapped with mock providers
 */
const TestProviders = ({ children }) => (
  <MockedProvider mocks={apolloMocks}>
    <ThemeProvider theme={defaultTheme}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </ThemeProvider>
  </MockedProvider>
);

TestProviders.propTypes = {
  /** React Component */
  children: PropTypes.element.isRequired
};

/**
 * Custom test renderer that wraps all components with the appropriate mock providers.
 * @param {Component} component - React component to render.
 * @param {Object} options - Options.
 * @return {Object} - @see {@link https://testing-library.com/docs/react-testing-library/api#render-result|react-testing-library}
 */
const renderWithProviders = (component, options) => (
  render(component, { wrapper: TestProviders, ...options })
);

export * from "@testing-library/react";
export { renderWithProviders as render };
