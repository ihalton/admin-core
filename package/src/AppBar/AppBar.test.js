import React from "react";
import { render } from "../test-utils";
import AppBar from "./AppBar";

test("render the AppBar with a title and children", () => {
  const { asFragment } = render((
    <AppBar title="Title">
      <span>Sub Component</span>
    </AppBar>
  ));
  expect(asFragment()).toMatchSnapshot();
});
