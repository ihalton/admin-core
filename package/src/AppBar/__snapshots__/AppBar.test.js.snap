// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the AppBar with a title and children 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation0"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-navigationDrawerOpen-2 MuiToolbar-gutters"
    >
      <h1
        class="MuiTypography-root makeStyles-title-4 MuiTypography-h3"
      >
        Title
      </h1>
      <div
        class="makeStyles-action-1"
      >
        <span>
          Sub Component
        </span>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
